### ISSUE TITLE
Preview mode selector requires scrolling when message is shown

### ISSUE THREAD

**Issue by th3hamm0r** [wagtail/wagtail#11609]
### Is your proposal related to a problem?

If a notification message is shown, e.g. when an object is locked, the preview-mode-selector seems to be displaced by the height of the message bar (multiple messages result in a larger displacement).
So if an object is locked, you always have to scroll down to change the preview mode.

[wagtail_preview_mode_selector.webm](https://github.com/wagtail/wagtail/assets/468164/10ae8c0e-0582-45e9-8cfb-caa56d1c5dd0)

### Describe the solution you'd like

It would be nice if the preview area would always be 100% of the height so that the selector remains accessible.

### Steps to Reproduce

1. Install the [bakerydemo](https://github.com/wagtail/bakerydemo)
2. Log into the CMS
3. Open the already locked person `/admin/snippets/base/person/edit/3/` (as in the video above)

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by laymonage** [comment]
Thank you for the report! I've been aware of this for a while but kept forgetting to create an issue/fix. And yes, I think this is something we need to fix 😄

**Comment by faishal882** [comment]
hey @th3hamm0r are you working on this issue, if not i would like to give a try

**Comment by th3hamm0r** [comment]
No, I'm not working on it, thanks for your initiative 👍

**Comment by input** [comment]
Hi all,
I can reproduce this issue using the bakerydemo and Wagtail 7.1a0, and am planning to work on it.

**Comment by input** [comment]
Having now had an initial look at this, I can think of a couple of potential solutions and would like to ask for advice from the core team about how best to proceed.

I should say that I'm new to core Wagtail development, so apologies if I don't yet fully understand the code(!), but here goes…

## Explanation

Essentially, the issue is caused by the CSS on the `<aside class="form-side">` element:
```
// abridged 
height: calc(100vh - 100%);
position: absolute;
top: 100%;
```

This positions the `aside` element relative to its parent (`<div class="w-sticky w-top-0 w-z-header">`) and visually directly below its sibling (`<header class="w-slim-header">`). In turn, the child preview area fills 100% of the available height and, therein, the preview controls (`<div class="w-preview__controls>`) are positioned at the end/bottom.

This all works fine when there are no messages, because the parent (`<div class="w-sticky w-top-0 w-z-header">`) element is positioned at the top of the viewport, and, hence, the calculation `height: calc(100vh - 100%);` fits everything in vertically.

However, when messages are displayed (`<div class="messages">`) they sit above the parent and push the parent down, which, in turn, moves the preview controls either partly or fully off-screen.


## Possible solutions 

Two simple solutions are:

1. Determine the amount of any offset and apply this to the `<aside class="form-side">` element.
(I notice that the minimap already appears to use offsetting, via a combination of JavaScript and CSS. So, perhaps a similar implementation could be used?).

2. Using CSS, reposition the preview controls at the top of the preview area. However, would a layout change be desirable/acceptable?

**Comment by mgax** [comment]
I've tested again using Wagtail `7.0.1` and the current `master` (20fe758c33) and the preview mode is at the bottom and doesn't scroll off screen. I suppose somebody fixed it already. 🤔 

<img width="400" alt="Screenshot" src="https://github.com/user-attachments/assets/5ccf5214-73dd-40f7-926b-ea1bb52323a5" />

**Comment by laymonage** [comment]
@mgax do you have a message banner at the top? This issue only happens when you have one (e.g. after saving as draft or if there's a validation error)

**Comment by mgax** [comment]
> [@mgax](https://github.com/mgax) do you have a message banner at the top? This issue only happens when you have one (e.g. after saving as draft or if there's a validation error)

Right, I missed that detail. With a message banner I get the same behaviour. I might have been confusing this with another issue, possibly on older Wagtail versions, where the selector required scrolling even with no banner (or maybe I'm simply misremembering).

**Comment by laymonage** [comment]
Thanks @input for the findings! We discussed this at the UI team meeting last week. I took another look at this and yes, I think the solution is to either move the preview mode options above the iframe, or use JavaScript to calculate and update the offset.

The latter seems a bit more complex. If I understand correctly, we want the iframe to take up whatever remaining space while still allowing the preview mode options to be immediately visible. This means the iframe will "grow"/"shrink" as you scroll down/up, so the result might be a bit jarring.

I think moving the preview mode selector to be above the iframe makes sense. This will also make the tab navigation order more consistent. We currently use `flex-direction: column-reverse` to swap the tab order so that tabbing from the preview size buttons will take you to the preview mode selector, before taking you to the iframe. This allows you to avoid tabbing through the whole iframe before being able to interact with the preview mode selector. However, this came at the cost of the inconsistent order compared to how the elements are presented.

We had designs that are somewhat like this when we put out the ["SEO power tools" call for sponsorship](https://wagtail.org/blog/looking-for-sponsorship-seo-power-tools/):

<img width="683" height="772" alt="Image" src="https://github.com/user-attachments/assets/9a4f567e-bfc6-4b50-a563-78d4bd26c514" />

https://www.figma.com/design/h67EsVXdWsfu38WGGxWfpi/Wagtail-CMS?node-id=19879-94507&t=XAL0ZQwkjCh4QSvp-4

Although I'm not sure if that design was meant to be final, as the Figma file seems to have different iterations of the concept in the context of SEO-related previews. If we want to go ahead with the above design, I think one thing to keep in mind is that the preview mode label can be quite long, so we need to make sure a long label is properly truncated instead of overflowing the container.
